gloo/types.h

```diff
@@ -56,17 +56,27 @@ struct __attribute__((__aligned__(2))) float16 {
     return x == res.x;
   }
 #ifdef __CUDA_ARCH__
-  float16(half h) {
+  float16(__half h) {
```

Comment:
Yeah, some more directed grepping found that for me too. I don't care which one is used